core/server/proxy/src/main/java/alluxio/proxy/s3/ListBucketResult.java

```diff
@@ -0,0 +1,394 @@
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Get bucket result defined in http://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html
+ * It will be encoded into an XML string to be returned as an error response for the REST call.
+ *
+ * TODO(chaomin): consider add more required fields in S3 BucketGet API.
+ */
+@JacksonXmlRootElement(localName = "ListBucketResult")
+@JsonPropertyOrder({ "Name", "Prefix", "ContinuationToken", "NextContinuationToken",
+    "KeyCount", "MaxKeys", "IsTruncated", "Contents" })
+public class ListBucketResult {
+
+  // Name of the bucket.
+  private String mName = "";
+  // Keys that begin with the indicated prefix.
+  private String mPrefix = null;
+  // ContinuationToken is included in the response if it was sent with the request.
+  private String mContinuationToken = null;
+  // Returns the number of keys included in the response. The value is always less than or equal
+  // to the mMaxKeys value.
+  private int mKeyCount = 0;
+  // The maximum number of keys returned in the response body.
+  private int mMaxKeys = S3RestUtils.S3_DEFAULT_MAX_KEYS;
+  // Specifies whether or not all of the results were returned. If the number of
+  // results exceeds that specified by MaxKeys, all of the results might not be returned.
+  private boolean mIsTruncated = false;
+  // If only partial results are returned (i.e. mIsTruncated = true), this value is set as the
+  // next continuation token. The continuation token is the full Alluxio path of the object.
+  private String mNextContinuationToken = null;
+  // A list of objects with metadata.
+  private List<Content> mContents = new ArrayList<>();
+
+  /**
+   * Creates an {@link ListBucketResult}.
+   */
+  public ListBucketResult() {}
+
+  /**
+   * Creates an {@link ListBucketResult}.
+   *
+   * @param bucketName the bucket name
+   * @param objectsList a list of {@link URIStatus}, representing the objects
+   * @param options the list bucket options
+   */
+  public ListBucketResult(
+      String bucketName, List<URIStatus> objectsList, ListBucketOptions options) {
+    mName = bucketName;
+    mPrefix = options.getPrefix();
+    mKeyCount = 0;
+    if (options.getMaxKeys() != null) {
+      mMaxKeys = Integer.parseInt(options.getMaxKeys());
+    }
+    mContents = new ArrayList<>();
+    mContinuationToken = options.getContinuationToken();
+
+    Collections.sort(objectsList, new URIStatusComparator());
+
+    int startIndex = 0;
+    if (options.getContinuationToken() != null) {
+      URIStatus tokenStatus = new URIStatus(new FileInfo().setPath(
+          mName + AlluxioURI.SEPARATOR + mContinuationToken));
+      startIndex = Collections.binarySearch(objectsList, tokenStatus, new URIStatusComparator());
+      if (startIndex < 0) {
+        startIndex = 0;
```

Comment:
Should there be a warning logged here?

Reply:
if this happens,  an `S3Error` should be responded

Reply:
We are supporting a non-existing `continuation-token` here, so binary searching's result at the 1st index should be acceptable.

Reply:
if my directory contains [a, c] and continuation-token=b => binary search returns -2. should startIndex be 1 here?

Reply:
oops, you are right, thanks!